### PR TITLE
Document YAML file with comments!

### LINKED ISSUE #3: YAML setup research
Document what each line means
Add tags where we need to change or update stuff

### PR DESCRIPTION
**Pull request by KrishnaSolo**
### Why
The YAML was a black box that needed to be documented. Added comments describing purpose
of a section and some important fields. Also highlighted fields we need to change.

### What
Added comments to yaml file.

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -33,7 +34,7 @@ Parameters:
     Default: prod
   CognitoDomainName:
     Type: String
-    Default: mytodoappdemo
+    Default: mytodoappdemo # need to update this
```
Comment (GurmeharS) [inline template.yaml:37]: lets just update these rn

File: template.yaml [hunk 2]
```diff
@@ -42,63 +43,64 @@ Parameters:
   Branch:
     Type: String
     Description: GitHub Branch
-    Default: master
+    Default: master # should be updated to main
```
Comment (GurmeharS) [inline template.yaml:46]: update this now as well

File: template.yaml [hunk 3]
```diff
@@ -42,63 +43,64 @@ Parameters:
...
   TodoUserPool:
-    Type: AWS::Cognito::UserPool
+    Type: AWS::Cognito::UserPool # read more - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html
     Properties:
       AdminCreateUserConfig:
-        AllowAdminCreateUserOnly: false
-      UserPoolName: TodoUsers
-      UsernameAttributes:
+        AllowAdminCreateUserOnly: false # this lets users sign themselves up
+      UserPoolName: TodoUsers # update name, also
```
Comment (GurmeharS) [inline template.yaml:60]: same as above

File: template.yaml [hunk 4]
```diff
@@ -42,63 +43,64 @@ Parameters:
...
           RequireSymbols: false
           RequireUppercase: true
   
-  TodoUserPoolTokenClient:
-    Type: AWS::Cognito::UserPoolClient
+  TodoUserPoolTokenClient: # rename
+    Type: AWS::Cognito::UserPoolClient # client is what our client side code will use - read more https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpoolclient.html
     Properties:
-      UserPoolId: !Ref TodoUserPool
+      UserPoolId: !Ref TodoUserPool # use our user pool made before - should be renamed
```
Comment (GurmeharS) [inline template.yaml:76]: Change name

File: template.yaml [hunk 5]
```diff
@@ -42,63 +43,64 @@ Parameters:
...
       ExplicitAuthFlows:
       - USER_PASSWORD_AUTH
   
-  TodoDomain:
+  TodoDomain: # create domain that hosts the sign-up and sign-in pages for your application - see more https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpooldomain.html
     Type: AWS::Cognito::UserPoolDomain
     Properties: 
       Domain: !Join ['-', [!Ref CognitoDomainName, !Ref AWS::StackName]]
-      UserPoolId: !Ref TodoUserPool
+      UserPoolId: !Ref TodoUserPool # change - link our user pool
```
Comment (GurmeharS) [inline template.yaml:91]: same as above

File: template.yaml [hunk 6]
```diff
@@ -109,71 +111,71 @@ Resources:
         SSEEnabled: true
 
   # API declaration
-  TodoApi:
-    Type: AWS::Serverless::Api
+  TodoApi: # this is implictly defined based on what functions we push - if we want to be explicit create a swagger definition of api
```
Comment (GurmeharS) [inline template.yaml:114]: Change name

File: template.yaml [hunk 7]
```diff
@@ -109,71 +111,71 @@ Resources:
...
-        LoggingLevel: INFO
+        ResourcePath: !Sub '${VersionParam}/*' # !sub is substitute - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html
+        LoggingLevel: INFO 
     AccessLogSetting:
       DestinationArn: !Sub 'arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:${ApiAccessLogGroup}'
       Format: '$context.identity.sourceIp $context.authorizer.claims.sub [$context.requestTime] "$context.httpMethod $context.resourcePath $context.protocol" $context.status $context.requestId $context.awsEndpointRequestId $context.xrayTraceId $context.responseLatency $context.integrationLatency "$context.error.message"'
     Properties:
-      Name: TodoApi
-      StageName: !Ref StageNameParam
+      Name: TodoApi # change
```
Comment (GurmeharS) [inline template.yaml:126]: Change now

File: template.yaml [hunk 8]
```diff
@@ -109,71 +111,71 @@ Resources:
...
       Cors:
         AllowOrigin: "'*'"
         AllowMethods: "'OPTIONS,HEAD,GET,PUT,POST,DELETE'"
         AllowHeaders: "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'"
-      Auth:
+      Auth: #  - https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#api-auth-object
         Authorizers:
           CognitoAuthorizer:
-            UserPoolArn: !GetAtt "TodoUserPool.Arn"
+            UserPoolArn: !GetAtt "TodoUserPool.Arn" # update this to link correct userpool id
```
Comment (GurmeharS) [inline template.yaml:136]: Just change this name for now as well, still need to look into what our link would be; should be similar regardless.

File: template.yaml [hunk 9]
```diff
@@ -109,71 +111,71 @@ Resources:
...
 
   # API Functions
   GetTodoFunction:
```
Comment (GurmeharS) [inline template.yaml:139]: Change name

**Review by GurmeharS — APPROVED** [review]
Just change all the common names for now. The rest we can change when we actually get there in setup

**Comment by GurmeharS** [comment]
Test

File: template.yaml (hunk 8 above)
Comment (KrishnaSolo) [reply template.yaml:136]: oh its what our user pool name is -> like above we instantiate the user pool called todouserpool - our name should be TakeNotesPool so the value is TakeNotesPool.Arn which gets the amazon resource number which links it

**Pull request closed without merging**